carma-messenger-core/cpp_message/include/cpp_message.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+/*
+ * Copyright (C) 2018-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+extern "C"
+{
+#include "MessageFrame.h"
+}
+
+#include <vector>
+#include <ros/ros.h>
+#include <carma_utils/CARMAUtils.h>
+#include <cav_msgs/ByteArray.h>
+#include <j2735_msgs/ControlRequest.h>
+
+namespace cpp_message
+{
+/**
+ * @class Message
+ * @brief Is the class responsible for DSRC message encoding and decoding
+ * 
+ * It currently only supports Geofence message type.
+ */
+class Message
+{
+private:
+
+    // node handles
+    std::shared_ptr<ros::CARMANodeHandle> nh_, pnh_;
+    
+    // ROS sub, pub and spin rate
+    int default_spin_rate_ = 10;
+    ros::Publisher outbound_binary_message_pub_;
+    ros::Subscriber outbound_geofence_request_message_sub_;
```

Comment:
I think we want an outbound and inbound for both the geofence message types. I have a suspicion we may have to use carma-messenger to the infrastructure interface to carma cloud so we will want both directions of data flow.

Reply:
The sub/pub for control message will be added by Saina later on.

Reply:
@qswawrq Ok so will that be a separate PR?